website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -5324,10 +5368,15 @@ resource utilization.
 A runner at any time may attempt to split a restriction while it is being processed. This allows the
 runner to either pause processing of the restriction so that other work may be done (common for
 unbounded restrictions to limit the amount of output and/or improve latency) or split the restriction
-into two pieces, increasing the available parallelism within the system. It is important to author a
-SDF with this in mind since the end of the restriction may change. Thus when writing the
-processing loop, it is important to use the result from trying to claim a piece of the restriction
-instead of assuming one can process till the end.
+into two pieces, increasing the available parallelism within the system. Please note that different
+runners(e.g., Dataflow, Flink, Spark) have different strategies to issue splits under batch and
+streaming execution.
+
+It is important to author an SDF with this in mind since the end of the restriction may change. Thus
```

Comment:
Cleaning up a little. WDYT about this?:

Author an SDF with this in mind since the end of the restriction may change. When writing the processing loop, use the result from trying to claim a piece of the restriction instead of assuming you can process until the end.